From ecaebdbcf693d9f08844e04dc97f6859b7c37c80 Mon Sep 17 00:00:00 2001
From: Andriy Gapon <avg@icyb.net.ua>
Date: Thu, 12 Nov 2015 11:23:37 -0800
Subject: [PATCH] OpenZFS 5778 - nvpair_type_is_array() does not recognize
 DATA_TYPE_INT8_ARRAY
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Authored by: Andriy Gapon <avg@icyb.net.ua>
Reviewed by: Matthew Ahrens <mahrens@delphix.com>
Reviewed by: Prakash Surya <prakash.surya@delphix.com>
Approved by: Dan McDonald <danmcd@omniti.com>
Reviewed-by: Don Brady <dev.fs.zfs@gmail.com>
Reviewed-by: George Melikov <mail@gmelikov.ru>
Reviewed-by: Brian Behlendorf <behlendorf1@llnl.gov>
Ported-by: Giuseppe Di Natale <dinatale2@llnl.gov>

OpenZFS-issue: https://www.illumos.org/issues/5778
OpenZFS-commit: https://github.com/openzfs/openzfs/commit/bf4d553
Closes #6580
---
 module/nvpair/nvpair.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/module/nvpair/nvpair.c b/module/nvpair/nvpair.c
index dffb226a2..24ef61e93 100644
--- a/module/nvpair/nvpair.c
+++ b/module/nvpair/nvpair.c
@@ -1244,6 +1244,7 @@ nvpair_type_is_array(nvpair_t *nvp)
 	data_type_t type = NVP_TYPE(nvp);
 
 	if ((type == DATA_TYPE_BYTE_ARRAY) ||
+	    (type == DATA_TYPE_INT8_ARRAY) ||
 	    (type == DATA_TYPE_UINT8_ARRAY) ||
 	    (type == DATA_TYPE_INT16_ARRAY) ||
 	    (type == DATA_TYPE_UINT16_ARRAY) ||
-- 
2.15.1

